lib/blocks/blockreader.go

```diff
@@ -21,13 +23,55 @@ import (
 
 var lineWithLeadingSpacesMatcher = regexp.MustCompile("^[[:space:]]*(.*\n)$")
 
-type blockReadFunc func(*Reader, int, string) error
+type blockReadFunc func(*Reader, int, string, bool) error
 
 type BlockWriter interface {
 	Write(index, startLine, endLine int, text string)
 	Close() error
 }
 
+type blockFormat interface {
```

Comment:
could we maybe move this into a file `textformat.go` ?

Reply:
:heavy_check_mark: